src/main/java/org/spongepowered/common/data/processor/data/item/BreakableDataProcessor.java

```diff
@@ -0,0 +1,118 @@
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.Sets;
+import net.minecraft.item.ItemStack;
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutableBreakableData;
+import org.spongepowered.api.data.manipulator.mutable.item.BreakableData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.common.Sponge;
+import org.spongepowered.common.data.manipulator.immutable.item.ImmutableSpongeBreakableData;
+import org.spongepowered.common.data.manipulator.mutable.item.SpongeBreakableData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+
+import java.util.List;
+import java.util.Set;
+
+public class BreakableDataProcessor extends AbstractSpongeDataProcessor<BreakableData, ImmutableBreakableData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof ItemStack;
+    }
+
+    @Override
+    public Optional<BreakableData> from(DataHolder dataHolder) {
+        if (dataHolder instanceof ItemStack) {
+            Optional<Set<BlockType>> blocks = dataHolder.get(Keys.BREAKABLE_BLOCK_TYPES);
+            if (blocks.isPresent()) {
+                return Optional.<BreakableData>of(new SpongeBreakableData(blocks.get()));
+            }
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<BreakableData> fill(DataHolder dataHolder, BreakableData manipulator, MergeFunction overlap) {
+        if (this.supports(dataHolder)) {
+            BreakableData merged = overlap.merge(checkNotNull(manipulator.copy()), this.from(dataHolder).get());
+            return Optional.of(manipulator.set(Keys.BREAKABLE_BLOCK_TYPES, merged.breakable().get()));
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<BreakableData> fill(DataContainer container, BreakableData breakableData) {
+        List<String> blockIds = container.getStringList(Keys.BREAKABLE_BLOCK_TYPES.getQuery()).get();
+        Set<BlockType> blockTypes = Sets.newHashSetWithExpectedSize(blockIds.size());
+        for (String blockId : blockIds) {
+            blockTypes.add(Sponge.getGame().getRegistry().getType(BlockType.class, blockId).get());
+        }
+        return Optional.of(breakableData.set(Keys.BREAKABLE_BLOCK_TYPES, blockTypes));
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, BreakableData manipulator, MergeFunction function) {
+        return dataHolder.offer(manipulator.breakable());
```

Comment:
We're already in a processor, why are you requesting more method calls to get the value processor?